daemon/config.go

```diff
@@ -75,6 +75,8 @@ func (config *Config) InstallFlags() {
 	config.Ulimits = make(map[string]*ulimit.Ulimit)
 	opts.UlimitMapVar(config.Ulimits, []string{"-default-ulimit"}, "Set default ulimits for containers")
 	flag.StringVar(&config.LogConfig.Type, []string{"-log-driver"}, "json-file", "Default driver for container logs")
+	config.LogConfig.Config = make(map[string]string)
```

Comment:
@LK4D4 map[string]string is a reference type. It is a pointer to the underlying map. When you pass it by value, the pointer value gets copied. If i allocate it after passing it by value, the copy gets allocated, not the original map. 
https://play.golang.org/p/HB4qUGYoGE